services/httpd/handler.go

```diff
@@ -376,7 +432,11 @@ func (h *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	w.Header().Add("X-Influxdb-Version", h.Version)
 	w.Header().Add("X-Influxdb-Build", h.BuildType)
 
-	if strings.HasPrefix(r.URL.Path, "/debug/pprof") && h.Config.PprofEnabled {
```

Comment:
I prefer to wrap the existing `handleProfiles` method, rather than adding to `h.mux`. Can you reuse the `authenticate()` func for this?

Reply:
I thought adding the authentication through `mux` would be cleaner than wrapping the handlers with `authenticate` here. Especially since the same wrapping pattern is used for `servePing`. Going through `AddRoutes` wraps a logging handler as well, which I thought might be useful.